integrationTests/ts/index.ts

```diff
@@ -2,17 +2,58 @@ import { GraphQLString, GraphQLSchema, GraphQLObjectType } from 'graphql/type';
 import { ExecutionResult } from 'graphql/execution';
 import { graphqlSync } from 'graphql';
 
+interface SomeExtension {
+  number: number;
+  string: string;
+}
+
+const example: SomeExtension = {
+  number: 42,
+  string: 'Meaning of life',
+};
+
+// The following code block requires a newer version of TypeScript
+/*! >=3.2
+
+declare module 'graphql' {
+  interface GraphQLObjectTypeExtensions<TSource = any, TContext = any> {
+    someObjectExtension?: SomeExtension;
+  }
+  interface GraphQLFieldExtensions<
+    TSource,
+    TContext,
+    TArgs = { [argName: string]: any }
+  > {
+    someFieldExtension?: SomeExtension;
+  }
+  interface GraphQLArgumentExtensions {
+    someArgumentExtension?: SomeExtension;
+  }
+}
+*/
```

Comment:
I don't know how best to selectively enable this for TypeScript 3.2 and above only.

Further, the types themselves are not concretely tested, pointing to an example in the GraphQL source where this takes place would be helpful.